src/dotnet-test-nunit/TestRunner.cs

```diff
@@ -236,6 +236,12 @@ void SetupLabelOutput(TestExecutionListener listener)
                  ? _options.DisplayTestLabels.ToUpperInvariant()
                  : "ON";
 
+            listener.TestStarted += (sender, args) =>
+            {
+                if (labels == "ALL")
```

Comment:
This will print the label twice, once when the test starts and once when the test finishes as per the TestFinished event below. NUnit prints the test label and the output at the end of each test to keep the labels in sync with the output when multiple tests are running in parallel. For example, if four test workers are running, you may see four labels printed, then the output for those four tests in whatever order they finish in making it impossible to match the output with the test.

It does make it difficult to determine which test is hanging, but people just need to know it is whatever test is **after** the last test printed. Since NUnit runs tests in alpha order internally, that usually isn't too hard.

The PCL version of the NUnit framework doesn't support multi-threading, but the next version of the framework will be netstandard and re-introduce parallel. I don't want to introduce incompatibilities here.

In the main NUnit framework, issue nunit/nunit#1139 was completed to allow you to output text from tests immediately (not wait until test finished) to further help developers diagnose failing tests. Instead of adding this `TestStarted` event, we should be making sure the pull request nunit/nunit#1536 for that issue works here.